### PR TITLE
Curbside Pullover Driving Logic in Simulation

### PR DESCRIPTION
**Pull request by vzyrianov**
- Driving Logic 
  - The Driving logic is currently setup to drive for 3 seconds, find a spot to pullover, start the pullover maneuver, and become `IDLE` (break) after reaching the target.
  - The point within the curbside furthest from other `Agents` is chosen as the pullover target. 

- Intent
  - The Klampt Visualization is updated to display the Intent in the top left corner and the pullover target.
  - The `intent` object is updated to provide the pullover target to other components. 
  - `PULL_OVER` and `PULL_OVER_PICKUP` Intents are added

- Logical Roadgraph Elements
  - The curbside `RoadgraphRegion` seen below is automatically created based on a curb `RoadgraphCurve` and width

 - Planning
   - A `PickupDropoffRoutePlanner` planner is created to handle the different Intents. 
   - For the pullover maneuver, a sigmoid-shaped motion is generated. This can be replaced by the A* planner in the future (I marked the area with a #TODO in route_planning.py).

https://github.com/krishauser/GEMstack/assets/15055342/042b9c74-d2d8-4227-900a-57da964b3e4c

### CODE REVIEW

**Review by krishauser — CHANGES_REQUESTED** [review]
Driving decisions are to be made in planning, not perception. Please move the files to the appropriate folder.

**Review by krishauser — CHANGES_REQUESTED** [review]
Intent logic does not belong in trajectory tracking. It should go into some motion planning component. Please do not modify the computation graph inputs for the trajectory tracker

**Comment by vzyrianov** [comment]
I moved the driving logic to planning and removed intent logic from trajectory tracking.

File: GEMstack/onboard/planning/pure_pursuit.py
Comment (krishauser) [inline GEMstack/onboard/planning/pure_pursuit.py:1]: Please revert this to the prior version

File: GEMstack/state/agent.py
Comment (krishauser) [inline GEMstack/state/agent.py:1]: Please revert this to the prior version

**Review by krishauser — CHANGES_REQUESTED** [review]
Looks good except for the two files that are barely changed. Please keep the existing versions

**Comment by vzyrianov** [comment]
The changes are reverted.

**Review by krishauser — APPROVED** [review]

**Pull request merged**